compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1529,6 +1550,25 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
 
     #[instrument(skip(self), level = "debug")]
     fn report_fulfillment_error(&self, error: &FulfillmentError<'tcx>) {
+        if self.tcx.sess.opts.unstable_opts.dump_solver_proof_tree
```

Comment:
can you deduplicate this into a helper fn or something? it's almost identical to the above.